packages/react-components/react-toast/src/state/vanilla/getPositionStyles.ts

```diff
@@ -1,21 +1,20 @@
 import { ToastOffsetObject, ToastOffset, ToastPosition } from '../types';
 
 interface PositionStyles {
-  position: 'fixed';
   top?: number;
   left?: number;
   right?: number;
   bottom?: number;
 }
 
 export const getPositionStyles = (position: ToastPosition, dir: 'rtl' | 'ltr', offset?: ToastOffset) => {
-  const positionStyles: PositionStyles = {
-    position: 'fixed',
-  };
+  const positionStyles: PositionStyles = {};
 
   const offsetStyles: ToastOffsetObject = offset ? (isShorthandOffset(offset) ? offset : offset[position] ?? {}) : {};
 
-  const { horizontal = 20, vertical = 16 } = offsetStyles;
+  const centered = position === 'top' || position === 'bottom';
+
+  const { horizontal = centered ? 0 : 20, vertical = 16 } = offsetStyles;
```

Comment:
Follow up from #29080 - There should be no default horizontal offset for a centered toast.

Adds tests to stop future regressions